src/IdealizedSimulations/neverworld_simulation.jl

```diff
@@ -0,0 +1,409 @@
+# Neverworld
+#
+# Ingredients:
+#
+#   * Zonally-periodic domain with continental shelves on all boundaries except Southern Ocean
+#       * longitude = (0, 60)
+#       * 2 configurations in latitude
+#           - Half basin: latitude = (-70, 0)
+#           - Full basin: latitude = (-70, 70)
+#       * z to -4000m
+#       * Southern Ocean channel from -60 to -40 (with a ridge to -2000 m)
+#
+#   * Zonally-homogeneous wind stress with mid-latitude jet and trade winds
+#
+#   * Buoyancy 
+#       * restoring hot at the equator and cold at the poles (parabola, cosine, smooth step function)
+#       * equator-pole buoyancy differential: 0.06 (α * g * 30 ≈ 0.06 with α=2e-4, g=9.81)
+#       * exponential initial vertical stratification with N² = 6e-5 and decay scale h = 1000 m
+#           - eg bᵢ = N² * h * exp(z / h)
+#
+#   * Quadratic bottom drag with drag_coefficient = 1e-3
+
+using CubicSplines
+
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Grids: xnode, ynode
+using Oceananigans.Operators: xspacing, yspacing
+using Oceananigans.Operators: Δzᶜᶜᶜ
+using Oceananigans.Coriolis: ActiveCellEnstrophyConservingScheme
+using Oceananigans.TurbulenceClosures: CATKEVerticalDiffusivity
+
+using ClimaOcean: CubicSplineFunction,
+                  u_bottom_drag, u_immersed_bottom_drag,
+                  v_bottom_drag, v_immersed_bottom_drag
+                  
+using ..VerticalGrids: stretched_vertical_cell_interfaces
+
+const c = Center()
+const f = Face()
+
+#####
+##### Utility
+#####
+
+instantiate(T::DataType) = T()
+instantiate(t) = t
+
+function minimum_spacing(dir, (LX, LY, LZ), grid)
+    spacing = eval(Symbol(dir, :spacing))
+    loc = map(instantiate, (LX, LY, LZ))
+    Δ = KernelFunctionOperation{LX, LY, LZ}(spacing, grid, loc...)
+    return minimum(Δ)
+end
```

Comment:
is this different from Oceananigans? Why not use the one from Oceananigans?

Reply:
good point, I think this was written before we had it in Oceanagnians